exodus_gw/routers/dynamodb.py

```diff
@@ -0,0 +1,113 @@
+import logging
+from typing import List, Tuple
+
+import backoff
+
+from ..aws.client import DynamoDBClientWrapper as ddb_client
+from ..settings import get_environment
+
+LOG = logging.getLogger("exodus-gw")
+
+
+@backoff.on_predicate(
+    backoff.expo,
+    lambda response: response["UnprocessedItems"],
```

Comment:
https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchGetItem.html#API_BatchGetItem_ResponseSyntax 

```suggestion
    lambda response: response["UnprocessedKeys"],
```

Reply:
The response this function returns (and therefore checks) is from `batch_write_item`. While the function does call `batch_get_item` to tell us which of the items are actually on the table, that response, seen filtered on line 48, is consumed by the `batch_write_item` call.

Reply:
Hmm, nevermind the above suggestion. I now see that this was referring to the response of batch_write_item. But, it's still possible for this function to return a batch_get_item response, which would result in a KeyError.

Reply:
Ah, yeah, I suppose if the `items` list resulting from the query was empty it would return the query's `reponse`. We definitely don't want that.  Thanks.

Reply:
Solved this by removing the query altogether, as Rohan suggested.